db/python/tables/family.py

```diff
@@ -167,14 +179,60 @@ async def get_family_external_ids_by_participant_ids(
     async def update_family(
         self,
         id_: int,
-        external_id: str | None = None,
+        external_ids: dict[str, str | None] | None = None,
         description: str | None = None,
         coded_phenotype: str | None = None,
     ) -> bool:
         """Update values for a family"""
-        values: Dict[str, Any] = {'audit_log_id': await self.audit_log_id()}
-        if external_id:
-            values['external_id'] = external_id
+        audit_log_id = await self.audit_log_id()
+        values: Dict[str, Any] = {'audit_log_id': audit_log_id}
+
+        if external_ids:
+            to_delete = [k.lower() for k, v in external_ids.items() if v is None]
+            to_update = {k.lower(): v for k, v in external_ids.items() if v is not None}
+
+            if to_delete:
+                # Set audit_log_id to this transaction before deleting the rows
+                _audit_update_query = """
+UPDATE family_external_id
```

Comment:
:shrug: There's a bit of a mixture of at-the-left-margin and inline-with-surrounding-code SQL in these files.

But mostly I was being lazy and trying to avoid excess line-wrapping… will revisit these and look at what's in the nearby code.